src/instrument_meta.py

```diff
@@ -0,0 +1,307 @@
+
+        do_update_stmt = insert_stmt.on_conflict_do_update(
+            constraint='unique_instrument_meta',
+            set_=dict(
+                num_channels = self.instrument_meta.num_channels,
+                scan_angle = self.instrument_meta.scan_angle,
+                updated_at = time_now
+            )
+        )
+
+        print(f'do update stmt: {do_update_stmt}')
+
+        try:
+            result = session.execute(do_update_stmt)
+            session.flush()
+            result_row = result.fetchone()
+            action = db_utils.INSERT
+            if result_row.updated_at is not None:
+                action = db_utils.UPDATE
+
+            session.commit()
+            session.close()
+        except Exception as err:
+            message = f'Attempt to INSERT/UPDATE instrument meta record FAILED'
+            error_msg = f'Failed to insert/update record - err: {err}'
+            print(f'error_msg: {error_msg}')
+            session.close()
+        else:
+            message = f'Attempt to {action} instrument meta record SUCCEEDED'
+            error_msg = None
+        
+        results = {}
+        if result_row is not None:
+            results['action'] = action
+            results['data'] = [result_row._mapping]
+            results['id'] = result_row.id
+
+        response = DbActionResponse(
+            self.request_dict,
+            (error_msg is None),
+            message,
+            results,
+            error_msg
+        )
+
+        print(f'response: {response}')
+        return response 
+    
+
+    def get_instrument_metas(self):
+        session = stm.get_session()
+
+        q = session.query(
+            im.id,
+            im.name,
+            im.num_channels,
+            im.scan_angle,
+            im.created_at,
+            im.updated_at
+        ).select_from(
+            im
+        )
+
+        print('Before adding filters to instrument meta request########################')
+        if self.filters is not None and len(self.filters) > 0:
+            for key, value in self.filters.items():
+                q = q.filter(value)
+        
+        print('After adding filters to instrument meta request########################')
+        
+        # add column ordering
+        column_ordering = db_utils.build_column_ordering(im, self.ordering)
+        if column_ordering is not None and len(column_ordering) > 0:
+            for ordering_item in column_ordering:
+                q = q.order_by(ordering_item)
+
+        # limit number of returned records
+        if self.record_limit is not None and self.record_limit > 0:
+            q = q.limit(self.record_limit)
+
+        instrument_metas = q.all()
+
+        results = DataFrame()
+        error_msg = None
+        record_count = 0
+        try:
+            if len(instrument_metas) > 0:
+                results = DataFrame(instrument_metas, columns = instrument_metas[0]._fields)
+            
+        except Exception as err:
```

Comment:
specific Exception?

Reply:
honestly just not sure on this one, but it's the same thing of if its an exception we want to mark that and pass it on so the code itself doesn't die on the user but we return a failure message